gradient_boosting.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+
+from sklearn.base import RegressorMixin
+from tree import DecisionTreeRegressor
+
+
+class BaseGradientBoostingRegressor:
+
+    def __init__(self,
+                 n_estimators: 5,
+                 max_depth: int,
+                 learning_rate=None):
+
+        self.estimator = DecisionTreeRegressor
+        self.n_estimators = n_estimators
+        self.max_depth = max_depth
+        self.learning_rate = [learning_rate] * n_estimators
```

Comment:
This behavior is not obvious. The user specifies `learning_rate` as `float` but inside it transforms into the `list()`. Please, create separate protected variable `_learning_rates_array` and use it for storing list of learning rates. But still, keep the variable `learning_rate` with the specified float value which is taken from the constructor.